### PR TITLE
Feature/2016 11 16 frederick bousson boozeometer

### PR DESCRIPTION
**Pull request by beele**

### CODE REVIEW

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+These applications show a full end-to-end implementation of the LoRa techonology leveraging the Proximus MyThings Internet of Things platform.
+
+<p style="text-align: center;">
+  <img class="image fit" style="height: 500px; margin:0px auto;" alt="Booze-o-meter at devoxx" src="../img/end-to-end-iot/booze-devoxx.jpg">
+</p>
+
+> The Booze-o-meter V2 at Devoxx Belgium 2016.
+
+# Building end-to-end LoRa Iot Solutions 
+Building an enterprise IoT solution is challanging. Devices need to be enrolled, monitored and maintained.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: challenging

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+
+## Technologies overview
+For our rapid prototypes and small to medium applications we have chosen the following technical stack:
+- Proximus LoRa network for LoRa connectivity
+- Proximus MyThings platform for device management
+- NodeJS for the backend with TypeScript for the backend
+- Angular for the frontend (The older versions are still on AngularJS)
+
+Below we will go into detail about each item shortly
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: Period at the end of the sentence.

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+<p style="text-align: center;">
+  <img class="image fit" style="height: 200px; margin:0px auto;" alt="LoRa" src="../img/end-to-end-iot/lora.png">
+</p>
+### LoRa
+
+LoRa, short for LoRaWAN is a LPWAN (Low Power Wide Area Network) meant for wireless battery powered devices of 'things'.
+It offers a low power, low bandwith secure network to transceive information across large distances. The network is layed out in star topology and can easily be extended by placing more base stations also called LoRa gateways.
+
+Some network parameters:
+- Range of 5 to 15 kilometers (3,1 to 9,3 miles) depending on the conditions and signal strenght.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: strength

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+
+<p style="text-align: center;">
+  <img class="image fit" style="width: 250px; margin:0px auto;" alt="NodeJS" src="../img/end-to-end-iot/node.png">
+</p>
+### Node.js &amp; TypeScript
+
+Most people should be familiar with [Node](https://nodejs.org/en/).
+It is the JavaScript runtime built upon the V8 engine that Google Chrome uses.
+It is a lightweigth and efficient runtimle that uses an event-driven, non-blocking I/O model.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: lightweight
runtime

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+This combined with the added type safety that TypeScript provides makes this an excellent choise for rapid prototyping.
+
+Some of our own demo applications make use of the Node Simple Server (NSS) application, while other just use express. This depends on the needs of the project.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: Express (capital)
Could also make it link to https://expressjs.com

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+If you are interested in the NSS project, we have a blogpost about it [here](http://ordina-jworks.github.io/iot/2017/01/21/Node-with-TypeScript.html) and it is [on github](https://github.com/ordina-jworks/NodeSimpleServer) too!
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: GitHub (capitals)

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+### Angular
+
+Like with Node, most people should be familiar with [Angular](https://angular.io) (or the older AngularJS).
+Angular is a development platform for building modern single page web applications.
+It is a complete rewrite of the older AngularJS and therefor has some big changes in how things work.
+Angular is easy to set up and use, it also is fully cross platform/browser compatible.
+
+
+## Our demo applications
+All our demo applications are publicly available in the [github project of NSS](https://github.com/ordina-jworks/NodeSimpleServer).
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: GitHub (capitals)

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+These demo application are ever evolving as we are currently porting them from the older AngularJS to Angular with TypeScript.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: applications

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+
+Our demo applications have been showcased and used at several events including internal Proximus events as well as conferences like Devoxx, Techorama and The Belgian IoT convention in Mechelen.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: Maybe make the conferences clickable?

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+
+Below we will go into detail about each application and how it came to be, as well as the iterations they went through.
+
+Aside from the Slotmachine and the Booze-o-meter we've also developed the Stairway to Health application for Proximus. 
+A blogpost about this will be available in October.
+For the impatiant, The IoT talk at the annual JWorks JOIN event convered this topic already and can be viewed on YouTube.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: impatient
... **t**he IoT talk ...
covered

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 11]
```diff
@@ -0,0 +1,188 @@
...
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/BbnwrvfozUs?rel=0" frameborder="0" allowfullscreen></iframe>
+
+
+### The slotmachine
+The slotmachine application does mostly what its name suggests, but with a twist.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: Slotmachine is written with a capital in certain places and without capital in other places.
Might be best to use to same notation everywhere.

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+
+The idea is simple:
+A simple push button sends a signal to the backend application.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: First step should be the user registering him/herself? :-)

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 13]
```diff
@@ -0,0 +1,188 @@
...
+The slotmachine application does mostly what its name suggests, but with a twist.
+
+The idea is simple:
+A simple push button sends a signal to the backend application. 
+The application dispatches an event via a websocket to the frontend application which turns the slotmachine. 
+The slotmachine can either result in a win or a loss. A maximum of three attempts are possible per player, after which a new user registration is required to play again.
+The user registration can be disabled depening on de requirements of the event/conference.
+
+The light effects are also controlled by the application. 
+If the user has registered the gentle fading switches to a running light effect and if the user wins the effect changes to a set of blinking colors!
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: ... user wins, the effect ... (comma)

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 14]
```diff
@@ -0,0 +1,188 @@
...
+The second version of the slotmachine application swapped out the Arduino and the required wired connection with a LoRa enabled push button.
+
+This allowed us to demonstrate the capabilities of the LoRa network in a fun and engaging way.
+The application remained unchanged for the user, and was adapted to be more configurable:
+Setting a win chance (up to 100%) and different images/styling for different events.
+
+
+### The booze-o-meter
+The Booze-o-meter application is a drink disperser that relays the remaining level of fluid in the dispenser.
+It is a fun example to demonstrate how measuring a liquid of fill level of a container can be achieved.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: Think it needs to be: "liquid fill level"?

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 15]
```diff
@@ -0,0 +1,188 @@
...
+This idea can be applied to container in a whole range of different industries and use cases.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: to a container?

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 16]
```diff
@@ -0,0 +1,188 @@
...
+
+> The Booze-o-meter V1 test setup with regular water.
+
+
+#### V1
+The first version of the Booze-o-meter used three sensors that can detect a liquid through a thin plastic container. This allowed us to present the level in the container in a coarse way: 
+- FULL (initial state)
+- HIGH (sensor)
+- MEDIUM (sensor)
+- FULL (sensor)
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: Two times FULL? I assume this must be "LOW"?

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 17]
```diff
@@ -0,0 +1,188 @@
...
+
+The sensors have a simple binary readout, true if liquid is detected, false if not.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: `true` (backticks)
`false` (backticks)

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 18]
```diff
@@ -0,0 +1,188 @@
...
+The second version of the Booze-o-meter application allowed us to get a more detailed reading of the remaining fluid level in the container thanks to the addition of an ultrasonic sensor. 
+This sensor can measure the distance between itself and a surface, in this case the surface of the liquid in the container.
+
+The application was updated to support this more granular approach that is able to show the level in the container accurate to 1%.
+
+
+## Conclusion
+
+Our demo applications have served us well in bringing across the idea of LoRa to customers and other interested developers. 
+We will continue to evolve our demo applications to add new features and keep them up to date with newer and experimental techonolgies.
```
Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: technologies

Comment (yannickdeturck) [inline _posts/2017-09-30-End-to-end-IoT.md]: You could add another small section at the end with all the useful links encountered in the blog post such as LoRa, MyThings and the GitHub repos.

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+
+<p style="text-align: center;">
+  <img class="image fit" style="height: 500px; margin:0px auto;" alt="Booze-o-meter at devoxx" src="../img/end-to-end-iot/booze-devoxx.jpg">
+</p>
+
+> The Booze-o-meter V2 at Devoxx Belgium 2016.
+
+# Building end-to-end LoRa Iot Solutions 
+Building an enterprise IoT solution is challanging. Devices need to be enrolled, monitored and maintained. 
+You can roll your own network and handle all of this yourself, this however will require quite the backend system to facilitate all of this.
```
Comment (andreasevers) [inline _posts/2017-09-30-End-to-end-IoT.md]: "roll your own" with "enrolled" before might be a slight style error.
"quite the backend system" seems a bit misused here

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+The Proximus LoRa network in combination with their MyThings platform takes away most of this and allows us to focus on the actual applications.
```
Comment (andreasevers) [inline _posts/2017-09-30-End-to-end-IoT.md]: Clarify "this" (stylewise)

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 21]
```diff
@@ -0,0 +1,188 @@
...
+
+
+## Technologies overview
+For our rapid prototypes and small to medium applications we have chosen the following technical stack:
+- Proximus LoRa network for LoRa connectivity
+- Proximus MyThings platform for device management
+- NodeJS for the backend with TypeScript for the backend
```
Comment (Turbots) [inline _posts/2017-09-30-End-to-end-IoT.md]: NodeJS with TypeScript **on** the backend

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 22]
```diff
@@ -0,0 +1,188 @@
...
+- Angular for the frontend (The older versions are still on AngularJS)
```
Comment (Turbots) [inline _posts/2017-09-30-End-to-end-IoT.md]: Angular **on** the frontend

File: _posts/2017-09-30-End-to-end-IoT.md (hunk 2 above)
Comment (Turbots) [reply _posts/2017-09-30-End-to-end-IoT.md]: We will look at each item in full detail below.

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 23]
```diff
@@ -0,0 +1,188 @@
...
+Below we will go into detail about each item shortly
+
+
+<p style="text-align: center;">
+  <img class="image fit" style="height: 200px; margin:0px auto;" alt="LoRa" src="../img/end-to-end-iot/lora.png">
+</p>
+### LoRa
+
+LoRa, short for LoRaWAN is a LPWAN (Low Power Wide Area Network) meant for wireless battery powered devices of 'things'.
+It offers a low power, low bandwith secure network to transceive information across large distances. The network is layed out in star topology and can easily be extended by placing more base stations also called LoRa gateways.
```
Comment (Turbots) [inline _posts/2017-09-30-End-to-end-IoT.md]: laid out*

Comment (Turbots) [reply _posts/2017-09-30-End-to-end-IoT.md]: more emphasis on key names, for example **LoRa Gateways**. dat leest makkelijker

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 24]
```diff
@@ -0,0 +1,188 @@
...
+
+<p style="text-align: center;">
+  <img class="image fit" style="width: 250px; margin:0px auto;" alt="NodeJS" src="../img/end-to-end-iot/node.png">
+</p>
+### Node.js &amp; TypeScript
+
+Most people should be familiar with [Node](https://nodejs.org/en/).
+It is the JavaScript runtime built upon the V8 engine that Google Chrome uses.
+It is a lightweigth and efficient runtimle that uses an event-driven, non-blocking I/O model.
+This combined with the added type safety that TypeScript provides makes this an excellent choise for rapid prototyping.
```
Comment (Turbots) [inline _posts/2017-09-30-End-to-end-IoT.md]: choice (met een C )

File: _posts/2017-09-30-End-to-end-IoT.md (hunk 9 above)
Comment (Turbots) [reply _posts/2017-09-30-End-to-end-IoT.md]: and more emphasis (bold)

File: _posts/2017-09-30-End-to-end-IoT.md [hunk 25]
```diff
@@ -0,0 +1,188 @@
...
+  <img class="image fit" style="width: 350px; margin:0px auto;" alt="Booze-o-meter application" src="../img/end-to-end-iot/booze-3.jpg">
+   <img class="image fit" style="width: 350px; margin:0px auto;" alt="Booze-o-meter application" src="../img/end-to-end-iot/booze-4.jpg">
+</p>
+
+> The Booze-o-meter V2 at Devoxx Belgium 2016 with actual liqour!
+
+The second version of the Booze-o-meter application allowed us to get a more detailed reading of the remaining fluid level in the container thanks to the addition of an ultrasonic sensor. 
+This sensor can measure the distance between itself and a surface, in this case the surface of the liquid in the container.
+
+The application was updated to support this more granular approach that is able to show the level in the container accurate to 1%.
```
Comment (Turbots) [inline _posts/2017-09-30-End-to-end-IoT.md]: accurately

File: _posts/2017-09-30-End-to-end-IoT.md (hunk 18 above)
Comment (Turbots) [reply _posts/2017-09-30-End-to-end-IoT.md]: ... by adding new features and keeping them up to date ...

**Comment by tomverelst** [comment]
Prepeand images with baseUrl

Example:

`Serverless`

**Pull request merged**